tsl/src/hypercore/vector_quals.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+#include <postgres.h>
+#include <utils/memutils.h>
+
+#include "arrow_tts.h"
+#include "vector_quals.h"
+
+/*
+ * Support functions to execute vectorized quals over arrow tuple table slots.
+ */
+
+/*
+ * Initialize the vector qual state.
+ */
+void
+vector_qual_state_init(VectorQualState *vqstate, List *quals, TupleTableSlot *slot)
+{
+	vqstate->vectorized_quals_constified = quals;
+	vqstate->per_vector_mcxt = arrow_slot_per_segment_memory_context(slot);
+	vqstate->get_arrow_array = vector_qual_state_get_arrow_array;
+	vqstate->num_results = TupIsNull(slot) ? 0 : arrow_slot_total_row_count(slot);
+	vqstate->slot = slot;
+}
```

Comment:
Nit: Might be a good idea to to a memset() to zero for good measure. Deals with those cases where you are missing initializing some parts of a struct and they get random data. Usually pretty cheap.

Reply:
Added a MemSet.